File: samples/SampleCounterV2/Mvu/Widgets/Button.cs
using Metano.TypeScript.DOM;

namespace SampleCounterV2.Mvu.Widgets;

public sealed class Button(string label, Action onPressed) : IWidget
{
    public HtmlElement Build()
    {
        var btn = Js.Document.CreateElement(HtmlElementType.Button);
        btn.TextContent = label;
        btn.OnClick = _ => onPressed();

        return btn;
    }
}

File: targets/js/sample-counter-v2/src/mvu/widgets/button.ts
import type { IWidget } from "#/mvu";

export class Button implements IWidget {
  private readonly _label: string;

  private readonly _onPressed: () => void;

  constructor(label: string, onPressed: () => void) {
    this._label = label;
    this._onPressed = onPressed;
  }

  build(): HTMLElement {
    const btn = document.createElement("button");
    btn.textContent = this._label;
    btn.onclick = (_) => this._onPressed();

    return btn;
  }
}
